### DISCUSSION TITLE
AccessToken validity

### DISCUSSION THREAD

**Discussion by sudhagar-mr-17051** [mastodon/mastodon#25522] [Q&A] [answered]
What is the validity of Access token generated using /oauth/token

will this vary for each mastodon instance

**Comment by andypiper** [comment, marked as answer]
Access tokens are specific to a Mastodon instance.

The only way that tokens become invalid is if a) the user deletes them from their account or b) the app revokes them.

I will look into adding this to the documentation for additional clarity, thanks for the great question here (but, please do not tag individuals for attention in discussions - where available and where appropriate, we will answer things as we see them - thank you!)

**Reply by sudhagar-mr-17051** [reply]
We are getting token revoked error even when the user changed the password of their mastodon account, is this expected? if so can you update the same in official documentation. SO that it will be clear for everyone trying to integrate mastodon via API.

If this is not an expected behaviour kindly let me know, so that we can check further for the actual cause of the issue.

**Reply by andypiper** [reply]
Can you confirm that this is a Mastodon server (not a similar service). I'm not able to immediately reproduce this myself - created an app on an instance running Mastodon 4.1.2, used the token to verify credentials via a REST call, changed the password for the account, and the same token continues to work as before (it was not revoked). So please check that there are no other factors involved there.

**Reply by jsobolenkova** [reply]
Hi! Could this issue be reopened? I am a user that encountered this issue today, googled & found this discussion. Got an error message during an attempt to edit my profile “The access token was revoked”. Profile is unable to use for now due to this issue.

**Reply by andypiper** [reply]
What version of Mastodon are you using, what instance? Could there have been any reason for the token to be revoked?

**Comment by zarch1972** [comment]
Just adding that I had exactly this too.  I changed the login password on the account, then straight away my code to post via the API started failing.

`mastodon.errors.MastodonUnauthorizedError: ('Mastodon API returned error', 401, 'Unauthorized', 'The access token was revoked')`

I refreshed the access token in the development settings, updated my code to use the new token and things started working again.

https://mastodonapp.uk/@energystatsuk/115031561631787038
https://energy-stats.uk/octopus-tracker-tariff-pricing/

Seems odd that the password is tied to the access token?
